### ISSUE TITLE
Unclear error message when selecting from table with empty schema

### ISSUE THREAD

**Issue by ongmingyang** [prestodb/presto#8654]
I'm using the mongodb connector to read data from mongodb. In the beginning, Presto inferred an empty schema because my table didn't have any data in it at that time. However, even after adding data, simple queries would continue to fail with a rather strange and uninformative error message:

```
presto:stuff> select * from table;
Query 20170803_032644_00003_2kmi5 failed: line 1:8: SELECT * not allowed in queries without FROM clause
com.facebook.presto.sql.analyzer.SemanticException: line 1:8: SELECT * not allowed in queries without FROM clause
```

Manually updating the schema in the `_schema` table solves the issue and the query works as expected. However, this error message is still strange.

I'm not very familiar with Presto so I would like to know if this is a bug or if it is intended. If it is intended, why?

**Comment by martint** [comment]
Yeah, it's a bug. The analyzer is making an assumption that tables have at least one field and that only queries without a from clause  (i.e., `SELECT *`) can have zero columns, so it's misreporting the error.

**Comment by kokosing** [comment]
I will take a look at it.

**Comment by kokosing** [comment]
Here is a fix: https://github.com/prestodb/presto/pull/8657, though I do not have an idea how to test without writing (changing existing) connector which could return such table.
